beacon-chain/db/validator.go

```diff
@@ -0,0 +1,74 @@
+package db
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+)
+
+// SaveValidatorIndex accepts a public key and validator index and writes them to disk.
+func (db *BeaconDB) SaveValidatorIndex(pubKey []byte, index int) error {
+	h := hashutil.Hash(pubKey)
```

Comment:
Why do we hash the pubkey?

Reply:
Yeah it's to save space, 96 bytes vs 32 bytes. That makes sense ya?

Reply:
Is it worth the performance cost though?

Reply:
It's about `1206 ns/op` over head with the additional hashing, but the trade off is key size goes down to 32 bytes as opposed 96 bytes

```go
func BenchmarkWithHash(b *testing.B) {
	for i := uint64(0); i < uint64(b.N); i++ {
		buf := make([]byte, binary.MaxVarintLen64)
		n := binary.PutUvarint(buf, i)
		_ = hashutil.Hash(buf[n:])
	}
}
```
`BenchmarkWithHash-4   	 1000000	      1206 ns/op`

Reply:
Thanks, I’m satisified with this answer